### PR TITLE
CapsUnlocked CU7

### PR DESCRIPTION
**Pull request by rys**
Support for the CapsUnlocked CU7, a 3x2 key macropad with clickable rotary encoder.

## Description

Basic support for the CU7, a 32u4-powered 6-key macropad with rotary encoder and RGB underglow. Tested on the production PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/capsunlocked/cu7/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+#define MEDIA_KEY_DELAY 10
```
Comment (drashna) [inline keyboards/capsunlocked/cu7/keymaps/default/keymap.c:19]:
```suggestion
```

File: keyboards/capsunlocked/cu7/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+// Volume up/down on the encoder
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  uint16_t held_keycode_timer = timer_read();
+
+  if (clockwise) {
+    register_code(KC_VOLU);
+    while (timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY) {}
+    unregister_code(KC_VOLU);
```
Comment (drashna) [inline keyboards/capsunlocked/cu7/keymaps/default/keymap.c:41]: `tap_code` handles this already, with the `TAP_CODE_DELAY` define. 
```suggestion
    tap_code(KC_VOLU);
```

File: keyboards/capsunlocked/cu7/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  }
+  else {
+    register_code(KC_VOLD);
+    while (timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY) {}
+    unregister_code(KC_VOLD);
```
Comment (drashna) [inline keyboards/capsunlocked/cu7/keymaps/default/keymap.c:46]:
```suggestion
    tap_code(KC_VOLD);
```

File: keyboards/capsunlocked/cu7/config.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#define RGB_DI_PIN F1
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+// #define RGBLIGHT_EFFECT_BREATHING
+#define RGBLED_NUM 6
+#define RGBLIGHT_HUE_STEP 10
+#define RGBLIGHT_SAT_STEP 17
+#define RGBLIGHT_VAL_STEP 17
+#define RGBLIGHT_LIMIT_VAL 50
+#endif
```
Comment (drashna) [inline keyboards/capsunlocked/cu7/config.h:72]:
```suggestion
#endif

#define TAP_CODE_DELAY 10
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/capsunlocked/cu7/rules.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no           # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no        # Unicode
+BLUETOOTH_ENABLE = no      # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no          # Audio output on port C6
+FAUXCLICKY_ENABLE = no     # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 	   # Enable support for HD44780 based LCDs (+400)
```
Comment (zvecr) [inline keyboards/capsunlocked/cu7/rules.mk:26]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/capsunlocked/cu7/keymaps/default/keymap.c (hunk 2 above)
Comment (fauxpark) [reply keyboards/capsunlocked/cu7/keymaps/default/keymap.c:41]: Can also do `tap_code_delay()` for a less global solution.

**Comment by rys** [comment]
Thanks for the review feedback, I've addressed everyting in 92d9174 (and I went with `tap_code_delay()`, didn't know that existed).

**Review by fauxpark — APPROVED** [review]

File: keyboards/capsunlocked/cu7/config.h [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ *
+*/
+
+#define ENCODER_RESOLUTION 2
+
+#define ENCODERS_PAD_A { D6 }
+#define ENCODERS_PAD_B { D4 }
+
+#define MATRIX_ROW_PINS { D7, F0, F6,  }
+#define MATRIX_COL_PINS { F5, F7, F4,  }
```
Comment (drashna) [inline keyboards/capsunlocked/cu7/config.h:49]:
```suggestion
#define MATRIX_ROW_PINS { D7, F0, F6 }
#define MATRIX_COL_PINS { F5, F7, F4 }
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**